### PR TITLE
Reintroduce ConnectionWrapper.wrap

### PR DESCRIPTION
**Pull request by felixbarny**
Fixes a breaking change in #400
Also concerns #401 

PS: We should probably agree on a common formatter or at least import order. Which IDE and formatter are you using? For p6spy, I'm using IntelliJ with default settings apart from indent=2 spaces.

### CODE REVIEW

**Comment by quintonm** [comment]
I am using IntelliJ. 

Most of the format settings should be controlled by [EditorConfig](http://editorconfig.org/) for anyone using IntelliJ.  Eclipse users will need a plugin though. It doesn't handle import statements though.

File: src/main/java/com/p6spy/engine/spy/P6SpyDriver.java [hunk 1]
```diff
@@ -18,29 +18,25 @@
...
-import java.sql.DriverManager;
-import java.sql.DriverPropertyInfo;
-import java.sql.SQLException;
-import java.sql.SQLFeatureNotSupportedException;
+import com.p6spy.engine.common.ConnectionInformation;
+import com.p6spy.engine.common.P6LogQuery;
+import com.p6spy.engine.event.JdbcEventListener;
+import com.p6spy.engine.wrapper.ConnectionWrapper;
+
+import java.sql.*;
```
Comment (quintonm) [inline src/main/java/com/p6spy/engine/spy/P6SpyDriver.java]: Opinion: Wildcard imports are bad.

**Comment by typekpb** [comment]
I'm using eclipse.

**Comment by felixbarny** [comment]
@typekpb are you also ok with the changes?

I don't think the ConnectionWrapper constructor should be public so that it's impossible that we end up with a `null` delegate and to make sure `onConnectionWrapped` is always called.

We could leave the non static `wrap` method and just return this to be more backwards compatible, but I'm not sure if it's worth it.

File: src/main/java/com/p6spy/engine/spy/P6SpyDriver.java (hunk 1 above)
Comment (typekpb) [reply src/main/java/com/p6spy/engine/spy/P6SpyDriver.java]: @quintonm agree

**Comment by typekpb** [comment]
@felixbarny LGTM (once at least one core reviewer agrees, you can merge, I guess. I would step up in case of objections)

File: src/main/java/com/p6spy/engine/spy/P6SpyDriver.java (hunk 1 above)
Comment (felixbarny) [reply src/main/java/com/p6spy/engine/spy/P6SpyDriver.java]: will fix

**Pull request merged**